### PR TITLE
[xc-admin] setup frontend template

### PR DESCRIPTION
**Pull request by cctdaniel**
- add xc-admin-frontend template
- add more placeholder stuffs
- update placefolders

### CODE REVIEW

File: xc-admin/packages/xc-admin-frontend/types/index.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { PriceComponent, PriceType, Product } from '@pythnetwork/client'
+import { PublicKey } from '@solana/web3.js'
+import { PriceStatus } from '../utils/PriceStatus'
+
+export interface PythPriceData {
```
Comment (guibescos) [inline xc-admin/packages/xc-admin-frontend/types/index.ts:5]: Aren't these types already in @pythnetwork/client?

File: xc-admin/packages/xc-admin-frontend/utils/pythClusterApiUrl.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { PythCluster } from '@pythnetwork/client/lib/cluster'
+
+const CLUSTER_URLS = {
```
Comment (guibescos) [inline xc-admin/packages/xc-admin-frontend/utils/pythClusterApiUrl.ts:3]: Make it  `Record<PythCluster, something>` for better types

File: xc-admin/packages/xc-admin-frontend/contexts/ClusterContext.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  setCluster: (cluster: PythCluster) => {},
+})
+
+export const ClusterProvider = (props: any) => {
+  const [cluster, setCluster] = useState<PythCluster>(DEFAULT_CLUSTER)
+  const contextValue = useMemo(
+    () => ({
+      cluster,
+      setCluster: (cluster: PythCluster) => {
+        const setCl = isValidCluster(cluster) ? cluster : DEFAULT_CLUSTER
```
Comment (guibescos) [inline xc-admin/packages/xc-admin-frontend/contexts/ClusterContext.tsx:22]: Why do you need to call `isValidCluster` on `cluster` which is already a `PythCluster`

File: xc-admin/packages/xc-admin-frontend/contexts/ClusterContext.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { Cluster } from '@solana/web3.js'
+import { createContext, useMemo, useState } from 'react'
+import { isValidCluster } from '../utils/isValidCluster'
+
+export type PythCluster = Cluster | 'pythtest' | 'pythnet'
```
Comment (guibescos) [inline xc-admin/packages/xc-admin-frontend/contexts/ClusterContext.tsx:5]: You can import this from @pythnetwork/client

File: xc-admin/packages/xc-admin-frontend/hooks/usePyth.ts [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+              return acc
+            }
+          })
+          .filter((_) => _) as AccountInfo<Buffer>[]
+    )
+    .flat()
+  return { keys, array }
+}
+
+export const ORACLE_PUBLIC_KEYS = {
```
Comment (guibescos) [inline xc-admin/packages/xc-admin-frontend/hooks/usePyth.ts:65]: Get this from @pythnetwork/client

**Review by guibescos — CHANGES_REQUESTED** [review]
Please fix CI

File: xc-admin/packages/xc-admin-frontend/hooks/usePyth.ts (hunk 5 above)
Comment (cctdaniel) [reply xc-admin/packages/xc-admin-frontend/hooks/usePyth.ts:65]: I don't see it in @pythnetwork/client

File: xc-admin/packages/xc-admin-frontend/types/index.ts (hunk 1 above)
Comment (cctdaniel) [reply xc-admin/packages/xc-admin-frontend/types/index.ts:5]: they are different tho

**Review by guibescos — APPROVED** [review]

**Pull request merged**